src/modules/Dropdown/Dropdown.js

```diff
@@ -458,6 +479,10 @@ export default class Dropdown extends Component {
     this.setState({
       selectedIndex: 0,
       searchQuery: newQuery,
+      addition: {
+        text: newQuery,
+        value: newQuery,
+      },
```

Comment:
When backspacing the search query, an empty option is left added to the dropdown menu:

![](http://g.recordit.co/4o2uXH7P6c.gif)

I'm thinking this feature may be better suited for a callback than state.  So, when a user attempts to add an item, we callback `onAddition` with the item value being added.  Then, the user can merge this item into the list of options themselves.  We would still fire `onChange` with addition value and also `trySetState` to add the addition to the dropdown value.

If we did this, all the mechanisms in place for getting items and setting the active item would work as is since the addition would simply be added to the actual list of options.  It would also give the developer opportunity to perform other tasks before allowing the item to be added.  For instance, they might want to disable the dropdown, show the loading state, and save the item back to the server.  Or, perform some client side validation before allowing the item in the options list.

Here is an example of what I am explaining:

```jsx
import React, { Component } from 'react'
import { Dropdown } from 'stardust'

export default class AllowAdditionsExample extends Component {
  state = { options: [{ text: 'English', value: 'English' }] }

  handleAddition = (value) => {
    // validate, save to server, show a msg to the user, etc.
    this.setState({
      options: [{ text: value, value }, ...this.state.options]
    })
  }

  render() {
    return (
      <Dropdown
        options={this.state.options}
        placeholder='Choose Language'
        search
        selection
        allowAdditions
        onAddition={this.handleAddition}
      />
    )
  }
}
```

This is not a full working example, but it illustrates the point.  Let me know your thoughts.